pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,222 @@
+        )
+
+    fm_deviation = Instrument.setting(
+        "FM %3.1fKZ",
+        """
+        Set the peak deviation in kHz for the FM function,
+        useable range 0.1 - 400 kHz
+
+        *NOTE*:
+            the maximum usable deviation is depending on the output frequency, refer to the
+            instrument documentation for further detail.
+
+        """,
+        validator=strict_range,
+        values=[0.1, 400],
+        )
+
+    fm_source = Instrument.setting(
+        "FM S%i",
+        """
+        Set the source for the FM function with :attr:`Modulation` enumeration.
+
+        ==========  =======
+        Value       Meaning
+        ==========  =======
+        OFF         no modulation active
+        INT_400HZ   internal 400 Hz modulation source
+        INT_1000HZ  internal 1000 Hz modulation source
+        EXTERNAL    External source, AC coupling
+        DC_FM       External source, DC coupling (FM only)
+        ==========  =======
+
+        *Note:*
+            * AM & FM can be active at the same time
+            * only one internal source can be active at the time
+            * use "OFF" to deactivate FM
+            * refer to the documentation rearding details on use of DC FM mode
+
+        usage example:
+
+        .. code-block:: python
+
+            sig_gen = HP8657B("GPIB::7")
+            ...
+            sig_gen.fm_source = sig_gen.Modulation.EXTERNAL     #  Enable external source for FM
+            sig_gen.fm_deviation = 15                           #  Set FM peak deviation to 15 kHz
+            ...
+            sig_gen.fm_source = sig_gen.Modulation.OFF          #  Turn FM off
+
+        """,
+        validator=strict_discrete_set,
+        values=Modulation,
+        )
+
+    frequency = Instrument.setting(
+        "FR %9.1f HZ",
+        """
+        Set the output frequency of the instrument in Hz.
+
+        For the 8567B the valid range is 100 kHz to 2060 MHz.
+        """,
+        validator=strict_range,
+        values=[1.0E5, 2.060E9],
+        )
+
+    level = Instrument.setting(
+        "AP %g DM",
+        """
+        Set the output level in dBm.
+
+        For the 8657B the range is -143.5 to +17 dBm/
+
+        """,
+        validator=strict_range,
+        values=[-143.5, 17.0],
+        )
+
+    level_offset = Instrument.setting(
+        "AO %g DB",
+        """
+        Set the output offset in dB, usable range -199 to +199 dB.
+
+        """,
+        validator=strict_range,
+        values=[-199.0, 199.0],
+        )
+
+    output_enabled = Instrument.setting(
+        "R%d",
+        """
+        Control whether the output is enabled.
+        """,
+        validator=strict_discrete_set,
+        values={False: 2, True: 3},
+        map_values=True
+       )
+
+    def reset(self):
+        self.adapter.connection.clear()
+
+    def shutdown(self):
+        self.output_enabled = False
+        self.adapter.connection.clear()
+        self.adapter.connection.close()
```

Comment:
Arguably, this should/could call `super().shutdown()`, too (although, cf #513). @bmoneke

Reply:
Oh, yes...

Reply:
Should I add this line `super().shutdown()`  and push again?
O should we  check all non-SCPI implementations and check if `shutdown()` has this line?

Reply:
The pull request is already merged. I can just add this line directly.

Reply:
Ok, thats also fine with me.
Thanks

Reply:
The PR has been merged, so you'd need to open another PR for that.
Arguably, all instruments that have redefined `shutdown` should call the parent method, too (or have a good/documented-by-comment reason not to). _That_ would be a useful PR, if you want to. :-)

Reply:
> The PR has been merged, so you'd need to open another PR for that. Arguably, all instruments that have redefined `shutdown` should call the parent method, too (or have a good/documented-by-comment reason not to). _That_ would be a useful PR, if you want to. :-)

A quick greo only gave about 20 instruments with `includeSCPI=False`, so Ill open a new PR for that